avail-intellij-plugin/src/main/kotlin/org/availlang/intellij/plugin/module/AvailProjectTemplate.kt

```diff
@@ -0,0 +1,244 @@
+import java.io.File
+
+/**
+ * The project template you get when you create a new Avail Project.
+ *
+ * Adapted to IDEA from [avail.anvil.manager.CreateProjectPanel]
+ *
+ * @author Raul Raja
+ *
+ */
+object AvailProjectTemplate {
+
+  /**
+   * Represents a configuration for an Avail software project.
+   *
+   * @property projectLocation The location of the project.
+   * @property fileName The name of the configuration file.
+   * @property rootsDir The directory where the project's root files are located.
+   * @property rootName The name of the project's root file.
+   * @property importStyles A flag indicating whether to import styles.
+   * @property importTemplates A flag indicating whether to import templates.
+   * @property libraryName The name of the library used in the project (optional).
+   * @property selectedLibrary The selected library file (optional).
+   * @property environmentSettings The global environment settings for the project.
+   */
+  data class Config(
+    val projectLocation: String,
+    val fileName: String,
+    val rootsDir: String,
+    val rootName: String,
+    val importStyles: Boolean,
+    val importTemplates: Boolean,
+    val libraryName: String?,
+    val selectedLibrary: File?,
+    val environmentSettings: GlobalEnvironmentSettings
+  )
+
+  /**
+   * Creates a new Avail project.
+   *
+   * @param config The configuration for the project.
+   */
+  @JvmStatic
+  fun create (config: Config)
+  {
+    val projectFilePath = "${config.projectLocation}/${config.fileName}.json"
+    val configPath =
+      AvailEnvironment.projectConfigPath(config.fileName, config.projectLocation)
+    AvailProject.optionallyInitializeConfigDirectory(configPath)
+    val localSettings = LocalSettings.from(File(configPath))
+    AvailProjectV1(
+      config.fileName,
+      true,
+      AvailRepositories(rootNameInJar = null),
+      localSettings
+    ).apply {
+      File(config.projectLocation).mkdirs()
+      val rootsDir = config.rootsDir
+      val rootName = config.rootName
+      val rootLocationDir = rootsLocationDir(rootsDir, rootName)
+      File("${config.projectLocation}/$rootLocationDir").mkdirs()
+
+      val rootConfigDir = AvailEnvironment.projectRootConfigPath(
+        config.fileName,
+        rootName,
+        config.projectLocation)
+      val root = availProjectRoot(rootConfigDir, config, rootName, rootLocationDir)
+      addRoot(root)
+      optionallyInitializeConfigDirectory(rootConfigDir)
+      config.selectedLibrary?.let { lib ->
+        configureAvailStdLib(config, lib)
+      }
+
+      updateBackingProjectFile(projectFilePath, config)
+    }
+  }
+
+  /**
+   * Updates the backing project file with the current AvailProjectV1 instance.
+   *
+   * @param projectFilePath The file path of the backing project file.
+   * @param config The configuration settings.
+   */
+  private fun AvailProjectV1.updateBackingProjectFile(
+    projectFilePath: String,
+    config: Config
+  )
+  {
```

Comment:
Move `)` and `{` to the same line. We used to be very particular about moving the `{` to its own line while keeping `)` on the line of the last argument, now we've taken to combining them on the next line if there 

1. There is more than one parameter
2. There is only one parameter but the whole function signature doesn't fit on one line
3. There is no return type

Here are examples:

```kotlin
fun fooAndSuch (
	qqqq: String,
	zzz: Double
){

////////////

fun fooAndSuch (
	qqqq: String,
	zzz: Double
): Baz
{

///////////

fun fooAndSuch (qqqq: String): Baz
{
///////////////////////

fun superReallyExtremlyVeryLongSignatureThatJustKeepsOnGoing(
	qqq: Long
): Baz
```